### DISCUSSION TITLE
Maven compiler plugin and Kotlin project

### DISCUSSION THREAD

**Discussion by hantsy** [quarkusio/quarkus#37902] [Community]
I am not sure this is related to Quarkus plugin.

When I updated my Quarkus example projects to use the latest maven compiler plugin 3.12.0 and Quarkus 3.6.4, all kotlin projects are failed to compile and I got the following error message:

```bash 
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.0:compile (default-compile) on project mongodb-kotlin: 
Fatal error compiling: basedir /home/runner/work/quarkus-sandbox/quarkus-sandbox/mongodb-kotlin/target/generated-sources/annotations does not exist -> [Help 1]
```

see: https://github.com/hantsy/quarkus-sandbox/actions/runs/7295787910/job/19882665078

**Comment by hantsy** [comment]
There is a `quarkus:generate-code` task, which does not the generate codes as expected???

**Reply by mschorsch** [reply]
There is a bug in the maven-compiler-plugin@3.12.0 (https://github.com/quarkusio/quarkus/pull/37852). Should be fixed in maven-compiler-plugin@3.12.1 but isnt't released yet.

**Reply by mschorsch** [reply]
maven-compiler-plugin@3.12.1 has been published.
